updater/lib/dependabot/dependency_change.rb

```diff
@@ -46,5 +46,51 @@ def updated_dependency_files_hash
     def grouped_update?
       !!dependency_group
     end
+
+    # This method combines checking the job's `updating_a_pull_request` flag
+    # with verification the dependencies involved remain the same.
+    #
+    # If the dependencies involved have changed, we should close the old PR
+    # rather than supersede it as the new changes don't necessarily follow
+    # from the previous ones; dependencies could have been removed from the
+    # project, or pinned by other changes.
+    def should_replace_existing_pr?
+      return false unless job.updating_a_pull_request?
+
+      # NOTE: Gradle, Maven and Nuget dependency names can be case-insensitive
+      # and the dependency name injected from a security advisory often doesn't
+      # match what users have specified in their manifest.
+      updated_dependencies.map(&:name).map(&:downcase) != job.dependencies.map(&:downcase)
```

Comment:
job.dependencies is a list of Dependabot::Dependency objects, so I think you would also need to map across the job dependency names

Reply:
This caught me out too, `dependency_snapshot.job_dependencies` are hydrated `Dependabot::Dependency` objects, but `job.dependencies` is just the [raw array from the job definition file](https://github.com/dependabot/dependabot-core/blob/9faebe9751245ea96a801007d86b96264eecd027/updater/spec/fixtures/job_definitions/bundler/version_updates/group_update_refresh.yaml#L10-L11).

We should probably do a naming cleanup pass on this to make that more obvious as I mocked tests incorrectly and then had a surprise when I tried it with the CLI.